src/main/java/org/folio/services/invoice/InvoiceCancelService.java

```diff
@@ -221,6 +221,10 @@ private Future<Void> unreleaseEncumbrancesForPoLines(List<PoLine> poLines, Invoi
     return encumbranceService.getEncumbrancesByPoLineIds(poLineIds, fiscalYearId, requestContext)
       .map(transactions -> transactions.stream()
         .filter(tr -> RELEASED.equals(tr.getEncumbrance().getStatus()))
+        // only unrelease encumbrances with expended + credited + awaiting payment = 0
+        .filter(tr -> tr.getEncumbrance().getAmountExpended() == 0
+          && tr.getEncumbrance().getAmountCredited() == 0
+          && tr.getEncumbrance().getAmountAwaitingPayment() == 0)
```

Comment:
I so we have the same logic in mod-orders, probably safer would be to create the same EncumbranceUtils even if we have usage only in 1 place here, in some future we need to think to move this Encumbrance utils to some shared library, maybe create additional subproject with jar executable in mod-finance